### DISCUSSION TITLE
Praise and a few feature suggestions

### DISCUSSION THREAD

**Discussion by someonesdad1** [sumatrapdfreader/sumatrapdf#2943] [Ideas]
**Praise**:  I read a suggestion to use SumatraPDF, tried it out, and in one day decided it was better than Foxit, my PDF client for the last couple of decades.  Sumatra has the lean features I want as a software developer; Foxit has turned into a corporate marketing mess with poor (Asian) support.  I copy/paste a lot from PDFs and Foxit's architecture absolutely sucks, making you go into a menu to select a different cursor.  Sumatra use the current item under the cursor to decide whether to scroll or highlight -- this is absolutely brilliant and, along with reload on file change (and not locking the damn file), is why I immediately switched to Sumatra.  

**Feature suggestions**

1. Add styles to the initialization file that I can select with some keyboard bindings I define.  I like both light and dark modes, but my preferred default is dark mode.  It looks like you complement or XOR the colors when I switch from white background to dark background; I'd love to have control over this, even to saying don't change any of the document's colors except the white & black.  Needs some prototyping.
2. Add a feature to show the page outline so page margins can be seen.  I'd also like to be able to see a line to show page demarcations; I saw this when I flipped white background to black (with white foreground), but not with the normal white background.

**Comment by GitHubRulesOK** [comment]
SumatraPDF locks files over 32MB but supposedly Foxit did not, so surprised that you say it does, but untested recently.

There are absolutely no current means to theme, system theming worked somewhat reasonable in XP, and attempts to part theme since proved unsuccessful, but 3rd party apps like windows blinds can change system features and colours.

1) Invert colors is designed for blueprints or negatives or inverted images thus full XOR colour reversal, (the paper has in many cases no colour or "white image pixels") ink colour is variable definitions, not simple to say all red objects or pixels to be black.

2) unclear of description but page gap is variable, in advanced settings.

**Reply by someonesdad1** [reply]
You can leave SumatraPDF alone and I'll still love it.  It's obviously written by programmers with an eye for programmers; AKA "I'm going to write what I want to use myself."  I worked at HP for decades and we called it "next bench" syndrome.  Congrats to your team for a sterling tool.  It's nearly as important to me as vim and a bash terminal.  Yeah, that important.  And I love the price.  :^)
